73_ArithmeticBench/main.cpp

```diff
@@ -0,0 +1,949 @@
+		};
+		m_device->updateDescriptorSets(1u, dsWrites, 0u, nullptr);
+
+		const uint32_t elementCount = Output<>::ScanElementCount;
+		const auto MinSubgroupSize = m_physicalDevice->getLimits().minSubgroupSize;
+		const auto MaxSubgroupSize = m_physicalDevice->getLimits().maxSubgroupSize;
+
+		const auto SubgroupSizeLog2 = hlsl::findMSB(MinSubgroupSize);
+
+		bool passed = true;
+		passed = runBenchmark<emulatedScanInclusive>(cmdbuf, benchSets[0], elementCount, SubgroupSizeLog2);
+		passed = runBenchmark<emulatedScanInclusive>(cmdbuf, benchSets[1], elementCount, SubgroupSizeLog2);
+		passed = runBenchmark<emulatedScanInclusive>(cmdbuf, benchSets[2], elementCount, SubgroupSizeLog2);
+
+
+		// blit
+		{
+			IGPUCommandBuffer::SPipelineBarrierDependencyInfo::image_barrier_t imageBarriers[2];
+			imageBarriers[0].barrier = {
+			   .dep = {
+				   .srcStageMask = PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT,
+				   .srcAccessMask = ACCESS_FLAGS::SHADER_WRITE_BITS,
+				   .dstStageMask = PIPELINE_STAGE_FLAGS::BLIT_BIT,
+				   .dstAccessMask = ACCESS_FLAGS::TRANSFER_WRITE_BIT
+				}
+			};
+			imageBarriers[0].image = dummyImg.get();
+			imageBarriers[0].subresourceRange = {
+				.aspectMask = IImage::EAF_COLOR_BIT,
+				.baseMipLevel = 0u,
+				.levelCount = 1u,
+				.baseArrayLayer = 0u,
+				.layerCount = 1u
+			};
+			imageBarriers[0].oldLayout = IImage::LAYOUT::UNDEFINED;
+			imageBarriers[0].newLayout = IImage::LAYOUT::TRANSFER_SRC_OPTIMAL;
+
+			imageBarriers[1].barrier = {
+			   .dep = {
+				   .srcStageMask = PIPELINE_STAGE_FLAGS::NONE,
+				   .srcAccessMask = ACCESS_FLAGS::NONE,
+				   .dstStageMask = PIPELINE_STAGE_FLAGS::BLIT_BIT,
+				   .dstAccessMask = ACCESS_FLAGS::TRANSFER_WRITE_BIT
+				}
+			};
+			imageBarriers[1].image = m_surface->getSwapchainResources()->getImage(m_currentImageAcquire.imageIndex);
+			imageBarriers[1].subresourceRange = {
+				.aspectMask = IImage::EAF_COLOR_BIT,
+				.baseMipLevel = 0u,
+				.levelCount = 1u,
+				.baseArrayLayer = 0u,
+				.layerCount = 1u
+			};
+			imageBarriers[1].oldLayout = IImage::LAYOUT::UNDEFINED;
+			imageBarriers[1].newLayout = IImage::LAYOUT::TRANSFER_DST_OPTIMAL;
+
+			cmdbuf->pipelineBarrier(E_DEPENDENCY_FLAGS::EDF_NONE, { .imgBarriers = imageBarriers });
+		}
+
+		{
+			IGPUCommandBuffer::SImageBlit regions[] = { {
+				.srcMinCoord = {0,0,0},
+				.srcMaxCoord = {WIN_W,WIN_H,1},
+				.dstMinCoord = {0,0,0},
+				.dstMaxCoord = {WIN_W,WIN_H,1},
+				.layerCount = 1,
+				.srcBaseLayer = 0,
+				.dstBaseLayer = 0,
+				.srcMipLevel = 0,
+				.dstMipLevel = 0,
+				.aspectMask = IGPUImage::E_ASPECT_FLAGS::EAF_COLOR_BIT
+			} };
+
+			auto srcImg = dummyImg.get();
+			auto scRes = static_cast<CDefaultSwapchainFramebuffers*>(m_surface->getSwapchainResources());
+			auto dstImg = scRes->getImage(m_currentImageAcquire.imageIndex);
+
+			cmdbuf->blitImage(srcImg, IImage::LAYOUT::TRANSFER_SRC_OPTIMAL, dstImg, IImage::LAYOUT::TRANSFER_DST_OPTIMAL, regions, ISampler::ETF_NEAREST);
+		}
```

Comment:
blit is not needed

Reply:
if you have the swapchain images bound and you layout transition them, Nsight should still consider them used, please remove the blit